opm/autodiff/BlackoilWellModel.hpp

```diff
@@ -84,6 +89,15 @@ namespace Opm {
             typedef Dune::FieldVector<Scalar, numEq    > VectorBlockType;
             typedef Dune::BlockVector<VectorBlockType> BVector;
 
+#if  DUNE_VERSION_NEWER_REV(DUNE_ISTL, 2 , 5, 1)
+            // 3x3 matrix block inversion was unstable from at least 2.3 until and
+            // including 2.5.0
+            typedef Dune::FieldMatrix<Scalar, numEq, numEq > MatrixBlockType;
+#else
```

Comment:
Also here: the dune version does not matter, but perhaps it should.

Reply:
Actually, the class used here should be the same as used in BlackoilModelEbos, so it should get it from there somehow. Easiest may be to just make it a template method, but that may not work for the well model classes that need to define a virtual interface.

Reply:
Gone in favor of using Dune::FMatrixHelper::invertMatrix